packages/x-date-pickers-pro/src/DateTimeRangePicker/shared.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+import { useThemeProps } from '@mui/material/styles';
+import { LocalizedComponent } from '@mui/x-date-pickers/locales';
+import {
+  DefaultizedProps,
+  useDefaultDates,
+  useUtils,
+  applyDefaultDate,
+  BaseDateValidationProps,
+  BasePickerInputProps,
+  PickerViewRendererLookup,
+} from '@mui/x-date-pickers/internals';
+import {
+  BaseClockProps,
+  DesktopOnlyTimePickerProps,
+} from '@mui/x-date-pickers/internals/models/props/clock';
+import { applyDefaultViewProps } from '@mui/x-date-pickers/internals/utils/views';
+import { TimeViewRendererProps } from '@mui/x-date-pickers/timeViewRenderers';
+import { TimeViewWithMeridiem } from '@mui/x-date-pickers/internals/models';
+import {
+  DigitalClockSlots,
+  DigitalClockSlotProps,
+} from '@mui/x-date-pickers/DigitalClock/DigitalClock.types';
+import {
+  MultiSectionDigitalClockSlots,
+  MultiSectionDigitalClockSlotProps,
+} from '@mui/x-date-pickers/MultiSectionDigitalClock/MultiSectionDigitalClock.types';
```

Comment:
These 2 interfaces should be exported from `'@mui/x-date-pickers/MultiSectionDigitalClock'`